libraries/tock-cells/src/map_cell.rs

```diff
@@ -17,23 +65,23 @@ pub struct MapCell<T> {
     // `.occupied` before calling `.val.get()` or `.val.assume_init()`. See
     // [mem::MaybeUninit](https://doc.rust-lang.org/core/mem/union.MaybeUninit.html).
     val: UnsafeCell<MaybeUninit<T>>,
```

Comment:
This does not drop a contained `T` and instead drops it early during `replace`

Reply:
Fixed.